source/sessions/tests/dirty-session-errors.yml

```diff
@@ -182,7 +192,7 @@ tests:
           - {_id: 2}
           - {_id: 3}
 
-  - description: Dirty implicit session is discarded
+  - description: Dirty implicit session is discarded (write)
```

Comment:
In "Dirty implicit session is discarded (read)", you use `times: 2` so that the test can pass w/ and w/o retryable reads. Here, you have `retryWrites: true` enabled in client options. Would it make sense to just disable retryable reads and writes for both tests instead?

I realize this test expects the insert to succeed based on the outcome assertion. Is my question above a non-concern because the retried insert (which succeeds) will use the same implicit LSID and it will only be marked dirty and discarded after the operation ultimately completes or fails?

Separate question: for bulk writes, I assume all writes in the bulk sequence need to use the same session, so the same rules would apply: retries or later commands in the same `bulkWrite` operation will all use the same LSID regardless of any network error encountered along the way (this probably also assumes `ordered: false` to allow for continue-on-error behavior). Perhaps this is covered by existing language you added to the spec: "It is valid for a dirty session to be used for subsequent operations (e.g. an implicit retry attempt)" -- if so, I might suggest adding "later commands in a bulk write" to the e.g. parenthetical.

Reply:
> Would it make sense to just disable retryable reads and writes for both tests instead?

Yes it would make sense but in order to disable retryable reads a driver has to implement retryable reads and I don't want to create that dependency for this spec change. Instead I choose to leave retryReads unspecified and make the read operation fail with/without retryable reads support by using `times:2`.

> Separate question: for bulk writes, I assume all writes in the bulk sequence need to use the same session

Yes that's also true, the latest commit expands the parenthetical.